### PR TITLE
fix: fix personal prices with unfulfilled orders

### PR DESCRIPTION
**Pull request by asadali145**
#### What are the relevant tickets?
https://github.com/mitodl/hq/issues/4043

#### What's this PR do?
Fixes a bug when PersonalPrice is created and unfulfilled orders exist with a greater amount.

#### How should this be manually tested?

1. Create an application for a boot camp.
2. Complete all the steps except the payment.
3. Make sure that the application state is `AWAITING_PAYMENT` at `/admin/applications/bootcampapplication/`
4. Click on make a payment and make a partial payment like **half** of the total amount. When redirected to CyberSource, just close the tab and do not complete the payment.
5. Now go to `/admin/klasses/personalprice/` and create a personal price with an amount less than the partial payment. i.e. Total BootCamp price was 1000 and you created an order of 500 then create a personal price of less than 500.
6. Go to the applications, your application state should be `AWAITING_PAYMENT`.
7. Now change the order status to `fulfilled` and change the personal price to a price lower than the order price.
8. Application should move to the fulfilled state now.

#### Where should the reviewer start?
Reproducing the Bug following the steps:
**Steps to reproduce:**

1. Create an application for a boot camp.
2. Complete all the steps except the payment.
3. Make sure that the application state is `AWAITING_PAYMENT` at `/admin/applications/bootcampapplication/`
4. Click on make a payment and make a partial payment like **half** of the total amount. When redirected to CyberSource, just close the tab and do not complete the payment.
5. Now go to `/admin/klasses/personalprice/` and create a personal price with an amount less than the partial payment. i.e. Total BootCamp price was 1000 and you created an order of 500 then create a personal price of less than 500.
6. Go to the applications, your application is moved to `COMPLETE` due to an unfulfilled order.

### CODE REVIEW

File: klasses/api_test.py [hunk 1]
```diff
@@ -174,6 +178,39 @@ def test_adjust_app_state_for_new_price(personal_price_amt, init_state, expected
...
+    user = app.user
+    personal_price = PersonalPriceFactory.create(price=990, user=user, bootcamp_run=run)
+    InstallmentFactory.create(bootcamp_run=run, amount=RUN_PRICE)
+    # Create payments such that the user has paid the original bootcamp run price
+    LineFactory.create_batch(
+        2,
+        order__user=user,
+        order__status=order_status,
+        bootcamp_run=run,
+        price=int(RUN_PRICE / 2),
```
Comment (Anas12091101) [inline klasses/api_test.py:200]: I think that this comment can be changed to something like 
`Create payments for the paid or pending order`

**Review by Anas12091101 — APPROVED** [review]
👍 Looks good. worked great. Just a small suggestion to change the comment in the test.

File: wire-transfers.csv
Comment (Anas12091101) [inline wire-transfers.csv:1]: What is the use of this file?

Comment (asadali145) [reply wire-transfers.csv:1]: It was an untracked file committed unintentionally. Thanks for identifying :)

File: klasses/api_test.py (hunk 1 above)
Comment (asadali145) [reply klasses/api_test.py]: Updated in https://github.com/mitodl/bootcamp-ecommerce/pull/1537/commits/3bb996c78a861b86aa9a7ebc2dd71c07f2714cdd

**Comment by asadali145** [comment]
@cachob Are we good to merge this change?

**Comment by cachob** [comment]
@asadali145 go ahead

**Pull request merged**